src/main/scala/org/allenai/pipeline/s3/S3Artifact.scala

```diff
@@ -116,20 +130,46 @@ trait S3Artifact extends Artifact with Logging {
 
   def readContents(): InputStream = service.getObject(bucket, path).getObjectContent
 
-  def download(file: File, bufferSize: Int = 1048576): Unit = {
+  def download(file: File): Unit = {
     logger.debug(s"Downloading $bucket/$path to $file")
     val tmpFile = new File(file.getCanonicalFile.getParent, file.getName + ".tmp")
-    val os = new FileOutputStream(tmpFile)
-    val is = readContents()
-    val buffer = new Array[Byte](bufferSize)
-    Iterator.continually(is.read(buffer)).takeWhile(_ != -1).foreach(n =>
-      os.write(buffer, 0, n))
-    is.close()
-    os.close()
-    require(tmpFile.renameTo(file), s"Unable to create $file")
+    tmpFile.deleteOnExit()
+    try {
+      Resource.using(readContents()) { is =>
+        Files.copy(is, tmpFile.toPath)
+      }
+      require(tmpFile.renameTo(file), s"Unable to create $file")
+    } finally {
+      tmpFile.delete() // just in case
+    }
```

Comment:
I didn't need to change this, but it now doesn't leak open streams even when exceptions are thrown, I could get rid of `bufferSize`, and I'm assuming that NIO (`Files.copy` is an NIO call) is faster than doing this ourselves.